### PR TITLE
Misc type stuff

### PR DESCRIPTION
**Pull request by nickdrozd**
## Description

Just some various type annotations.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

### CODE REVIEW

**Review by Pierre-Sassoulas — APPROVED** [review]
👌

File: astroid/_ast.py [hunk 1]
```diff
@@ -36,6 +37,7 @@ class ParserModule(NamedTuple):
     context_classes: dict[type[ast.expr_context], Context]
 
     def parse(self, string: str, type_comments: bool = True) -> ast.Module:
+        parse_func: Callable[[str], ast.Module]
```
Comment (DanielNoord) [inline astroid/_ast.py:40]: This isn't really correct with the `partial`, but let's use it for now. We're removing/revisiting most of this code anyway when we drop 3.7.

File: astroid/brain/brain_uuid.py [hunk 2]
```diff
@@ -8,7 +8,7 @@
 from astroid.nodes.scoped_nodes import ClassDef
 
 
-def _patch_uuid_class(node):
+def _patch_uuid_class(node) -> None:
```
Comment (DanielNoord) [inline astroid/brain/brain_uuid.py:11]:
```suggestion
def _patch_uuid_class(node: nodes.ClassDef) -> None:
```

File: astroid/brain/helpers.py [hunk 3]
```diff
@@ -5,8 +5,8 @@
 from astroid.nodes.scoped_nodes import Module
 
 
-def register_module_extender(manager, module_name, get_extension_mod):
-    def transform(node):
+def register_module_extender(manager, module_name: str, get_extension_mod) -> None:
+    def transform(node) -> None:
```
Comment (DanielNoord) [inline astroid/brain/helpers.py:9]:
```suggestion
    def transform(node: nodes.Module) -> None:
```

File: astroid/brain/helpers.py [hunk 4]
```diff
@@ -5,8 +5,8 @@
 from astroid.nodes.scoped_nodes import Module
 
 
-def register_module_extender(manager, module_name, get_extension_mod):
-    def transform(node):
+def register_module_extender(manager, module_name: str, get_extension_mod) -> None:
```
Comment (DanielNoord) [inline astroid/brain/helpers.py:8]:
```suggestion
def register_module_extender(manager: AstroidManager, module_name: str, get_extension_mod: Callable[[], nodes.Module]) -> None:
```

File: astroid/manager.py [hunk 5]
```diff
@@ -78,12 +78,12 @@ def __init__(self) -> None:
         self._transform = AstroidManager.brain["_transform"]
 
     @property
-    def register_transform(self):
+    def register_transform(self) -> Callable[[Any, Any, Any], None]:
```
Comment (DanielNoord) [inline astroid/manager.py:81]: Let's not do this. We're actually losing information here as the current return type seen in IDEs will include the default `None` value for the third argument.

File: astroid/manager.py [hunk 6]
```diff
@@ -78,12 +78,12 @@ def __init__(self) -> None:
...
         # This and unregister_transform below are exported for convenience
         return self._transform.register_transform
 
     @property
-    def unregister_transform(self):
+    def unregister_transform(self) -> Callable[[Any, Any, Any], None]:
```
Comment (DanielNoord) [inline astroid/manager.py:86]: Same here.

File: astroid/util.py [hunk 7]
```diff
@@ -22,7 +22,7 @@ def __get__(self, instance, owner=None):
     return DescriptorProxy(obj)
 
 
-def lazy_import(module_name):
+def lazy_import(module_name: str):
```
Comment (DanielNoord) [inline astroid/util.py:25]:
```suggestion
def lazy_import(module_name: str) -> Proxy:
```
?

File: astroid/util.py [hunk 8]
```diff
@@ -37,7 +37,7 @@ def __repr__(self) -> str:
 
     __str__ = __repr__
 
-    def __getattribute__(self, name):
+    def __getattribute__(self, name: str):
```
Comment (DanielNoord) [inline astroid/util.py:40]:
```suggestion
    def __getattribute__(self, name: str) -> Any:
```

**Review by DanielNoord — CHANGES_REQUESTED** [review]
Made some suggestions for some the types!

File: astroid/_ast.py (hunk 1 above)
Comment (nickdrozd) [reply astroid/_ast.py:40]: What should it be instead? Isn't it a function that takes a string and gives back a module?

Comment (DanielNoord) [reply astroid/_ast.py:40]: Well `partial` is its one type I believe, but it should do for now!

File: astroid/brain/helpers.py (hunk 3 above)
Comment (DanielNoord) [reply astroid/brain/helpers.py:16]: Could you do this suggestion as well?

**Review by DanielNoord — CHANGES_REQUESTED** [review]
One final suggestion!

**Review by Pierre-Sassoulas — APPROVED** [review]
👍

**Review by DanielNoord — APPROVED** [review]

**Comment by DanielNoord** [comment]
@Pierre-Sassoulas Again a failing CI.. Any idea?

**Comment by Pierre-Sassoulas** [comment]
This one we need to relaunch until it work, I guess it's github actions acting up.

**Pull request merged**